src-electron/util/util.js

```diff
@@ -147,14 +147,16 @@ async function initializeSessionPackage(db, sessionId, options) {
                     optionDefault.optionRef
                   )
                   .then((option) => {
-                    querySession.setSessionClean(db, sessionId)
                     return querySession.insertSessionKeyValue(
                       db,
                       sessionId,
                       option.optionCategory,
                       option.optionCode
                     )
                   })
+                  .then(async () => {
+                    await querySession.setSessionClean(db, sessionId)
```

Comment:
You can just do: 
```
   .then( () => querySession.setSessionClean(...)  )
```
without the squiggly brackets. That will return the promise from there to the upper chain.

One request:
   can you unroll this whole block into await?
```
 return Promise.all(promises)
    .then(() => queryPackage.getSessionPackages(db, sessionId))
    .then((packages) => {
      let p = packages.map((pkg) =>
        queryPackage
          .selectAllDefaultOptions(db, pkg.packageRef)
          .then((optionDefaultsArray) =>
            Promise.all(
              optionDefaultsArray.map((optionDefault) => {
                return queryPackage
                  .selectOptionValueByOptionDefaultId(
                    db,
                    optionDefault.optionRef
                  )
                  .then((option) => {
                    return querySession.insertSessionKeyValue(
                      db,
                      sessionId,
                      option.optionCategory,
                      option.optionCode
                    )
                  })
              })
            )
          )
      )
      return Promise.all(p).then(() => packages)
    })
```

These are so clumsy. You just turn this into kinda;
```
await Promise.all(promises)
let packages = await queryPackage.getSessionPacakges(db,sessionId)
.
.
.etc
```

We used to write these long Promise.....then().....then() constructs.
Lately we all switched to doing async/await patter, because it's much more readable and in 99% of the cases they are easier to maintain.
```

Reply:
Okay, thanks for the info. I am on it.